EventSourceProxy/TraceMethodAttribute.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+#if NUGET
+namespace EventSourceProxy.NuGet
+#else
+namespace EventSourceProxy
+#endif
+{
+	/// <summary>
+	/// Allows a method to be executed against a value before tracing.
+	/// </summary>
+	/// <remarks>Consumers can derive from this base to create their own methods to run against values being traced.
+	/// This is particularly useful as a means of masking or filtering data.
+	/// </remarks>
+	[AttributeUsage(AttributeTargets.Parameter, AllowMultiple = true)]
+	public abstract class TraceMethodAttribute : TraceAsAttribute
```

Comment:
What do you think about calling this TraceTransform/TraceTransformAttribute? TraceMethod is a little ambiguous (i.e. one might assume this attribute helps them trace a method, rather than providing a method that helps them trace a parameter). Just a suggestion.